src/SprykerEco/Zed/Payolution/Business/Payment/Method/AbstractPaymentMethod.php

```diff
@@ -180,11 +182,31 @@ protected function formatAddress($addressTransfer)
 
     /**
      * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     * @param array $salesOrderItems
      *
      * @return int
      */
-    protected function getGrandTotal(OrderTransfer $orderTransfer)
+    protected function getGrandTotal(OrderTransfer $orderTransfer, array $salesOrderItems)
     {
+        $idSalesOrderItems = [];
+
+        /** @var \Orm\Zed\Sales\Persistence\SpySalesOrderItem $salesOrderItem */
+        foreach ($salesOrderItems as $salesOrderItem) {
+            $idSalesOrderItems[] = $salesOrderItem->getIdSalesOrderItem();
+        }
+
+        if (!empty($idSalesOrderItems)) {
+            $grandTotal = 0;
+
+            foreach ($orderTransfer->getItems() as $item) {
+                if (in_array($item->getIdSalesOrderItem(), $idSalesOrderItems)) {
+                    $grandTotal += $item->getUnitPriceToPayAggregation();
```

Comment:
I think you should use `sumPriceToPayAggregation` here as we may have nonsplit items in order. For example if you buy product with quantity greater than 10.